scripts/remove-unused-locales.ts

```diff
@@ -0,0 +1,204 @@
+    })
+    .filter(d => d.isDirectory());
+  const localeNamesArr = localeFolders.map(d => d.name);
+  console.log(
+    chalk.green(
+      `found ${localeNamesArr.length} locales in the project: ${localeNamesArr}`
+    )
+  );
+  return localeNamesArr;
+};
+
+/**
+ *  self explanatory, uses yq to delete a key from a locale
+ * @param entry the key to delete
+ * @param localeName the locale to delete the key from,
+ * e.g. 'en' or 'fr'
+ */
+const deleteYmlEntry = (entry: string, localeName: string) =>
+  new Promise(res =>
+    exec(
+      `yq -i 'del(.${entry})' ${path.join(
+        path.join(__dirname, "../locales"),
+        localeName,
+        "index.yml"
+      )}`,
+      (err, _stdout, _) => {
+        if (err) {
+          console.log(chalk.red(`error deleting ${entry}`));
+          return res(false);
+        } else {
+          console.log(chalk.grey(`deleted ${entry}`));
+          return res(true);
+        }
+      }
+    )
+  );
+
+/**
+ * uses yq to look for null or undefined entries, skippable
+ * @param locale  the locale to clean up, e.g. 'en' or 'fr'
+ * @returns  a promise that resolves to true if the cleanup was successful
+ */
+const cleanupLocales = async (locale: string) =>
+  new Promise(res =>
+    exec(
+      `yq -i " del(.. |select(.==null or .== {})) " ` +
+        path.join(__dirname, "../locales", locale, "index.yml"),
+      (err, _stdout, _) => {
+        if (err) {
+          console.log(chalk.red(`${err} error cleaning up ${locale}`));
+          return res(false);
+        } else {
+          console.log(chalk.green(`cleaned up ${locale} locale`));
+          return res(true);
+        }
+      }
+    )
+  );
+
+const findUnused = async (root: string, locale: string) => {
+  const d = await readLocaleDoc(root, locale);
+  const keys = extractKeys(d.doc);
+  const unusedPromises = await Promise.all(keys.map(k => isUnused(k)));
+  return keys.filter((_, idx) => unusedPromises[idx]);
+};
+
+const main = async (root: string, locale: string) => {
+  const localeLangNames = getAllLocaleNames();
+  const unusedKeys = await findUnused(root, locale);
+
+  if (unusedKeys.length === 0) {
+    console.log(chalk.green("no unused keys found!"));
+    process.exit(1);
+  }
+  console.log(chalk.bgBlue("checking for yq installation..."));
+  await installYqIfNeeded();
+  console.log(chalk.red(`found ${unusedKeys.length} unused keys, deleting...`));
+  // for every locale, delete every unused key
+  for (const localeName of localeLangNames) {
+    console.log(
+      chalk.green(
+        `deleting unused keys from ${localeName.toUpperCase()} locale`
+      )
+    );
+    for (const k of unusedKeys) {
+      await deleteYmlEntry(k, localeName);
+    }
+    // do something to clean locale up, in order to avoid null or empty objects
+    await cleanupLocales(localeName);
+  }
+};
+
+main(path.join(__dirname, "../locales"), "it").then(
+  () => {
+    console.log("done");
+  },
+  () => process.exit(1)
+);
```

Comment:
This works only for IT locale. Wouldn't be better to pass the locale as parameter to the script?

Reply:
with the structure we are using for I18n( it being 
IT-> main locale
EN-> secondary locale
[??]-> tertiaries)
main and secondary locales have to be synced, so they have the same number of keys, while the tertiaries *may* have the same keys, but most do not appear.

That said, I believe adding an argument for the locale would simply move the locale param from the script to the node-modules, so it would really have no sense imho.